src/pages/People/PeoplePage.tsx

```diff
@@ -0,0 +1,131 @@
+import { useContext, useEffect } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { Loader } from '../../components/Loader';
+import { PeopleList } from '../../components/PeopleList/PeopleList';
+import { PeopleContext } from '../../peopleContext';
+import { PeopleFilters } from '../../components/PeopleFilters/PeopleFilters';
+import { getSearchWith } from '../../utils/searchHelper';
+import { SortingFields } from '../../types/SortingFields';
+import classNames from 'classnames';
+
+export const PeoplePage = () => {
+  const [searchParams] = useSearchParams();
+  const {
+    loader,
+    warning,
+    setLoader,
+    setPeople,
+    setWarning,
+    setCenturiesArray,
+  } = useContext(PeopleContext);
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  useEffect(() => {
+    if (warning) {
+      setWarning('');
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchParams]);
+
+  useEffect(() => {
+    getPeople()
+      .then(res => {
+        if (!res.length) {
+          setWarning('There are no people on the server');
+        } else {
+          setWarning('');
+        }
+
+        setPeople([...res]);
+        let tempArr: string[] = [];
+
+        res.forEach(item => {
+          if (!tempArr.includes((Math.floor(item.born / 100) + 1).toString())) {
+            tempArr = [
+              ...tempArr,
+              (Math.floor(item.born / 100) + 1).toString(),
+            ];
+          }
+        });
+        setCenturiesArray(tempArr.sort((a, b) => +a - +b));
+      })
+      .catch(error => {
+        setWarning('Something went wrong');
+        throw error;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)